### PR TITLE
Feature/hw 66553 58 62 consolidated user endpoint java

### PR DESCRIPTION
**Pull request by sgupta-hw-dev**
Create/Update User Changes

### CODE REVIEW

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 1]
```diff
@@ -24,6 +27,7 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
 import static org.testng.Assert.assertTrue;
+//import static org.hamcrest.Matchers.nullValue;
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:30]: Please remove the commented line

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 2]
```diff
@@ -128,17 +136,25 @@ public void testCreateUser_withoutProgramToken() throws Exception {
...
         HyperwalletUser apiClientUser = argument.getValue();
         assertThat(apiClientUser, is(notNullValue()));
         assertThat(apiClientUser.getFirstName(), is(equalTo("test-first-name")));
         assertThat(apiClientUser.getBusinessOperatingName(), is(equalTo("test-business-operating-name")));
         assertThat(apiClientUser.getStatus(), is(nullValue()));
-        assertThat(apiClientUser.getVerificationStatus(), is(VerificationStatus.VERIFIED));
+        assertThat(apiClientUser.getVerificationStatus(), is(equalTo(VerificationStatus.VERIFIED)));
+        assertThat(apiClientUser.getLetterOfAuthorizationStatus(), is(equalTo(LetterOfAuthorizationStatus.VERIFIED)));
+        assertThat(apiClientUser.getBusinessStakeholderVerificationStatus(), is(equalTo(BusinessStakeholderVerificationStatus.VERIFIED)));
+        assertThat(apiClientUser.getLetterOfAuthorizationStatus(), is(equalTo(LetterOfAuthorizationStatus.VERIFIED)));
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:156]: above line repeated

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 3]
```diff
@@ -162,15 +178,20 @@ public void testCreateUser_withProgramTokenAddedByDefault() throws Exception {
...
         assertThat(apiClientUser.getBusinessOperatingName(), is(equalTo("test-business-operating-name")));
         assertThat(apiClientUser.getStatus(), is(nullValue()));
         assertThat(apiClientUser.getVerificationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getBusinessStakeholderVerificationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getLetterOfAuthorizationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getGovernmentIdType(), is(nullValue()));
+        assertThat(apiClientUser.getBusinessStakeholderVerificationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getLetterOfAuthorizationStatus(), is(nullValue()));
         assertThat(apiClientUser.getCreatedOn(), is(nullValue()));
         assertThat(apiClientUser.getProgramToken(), is(equalTo("test-program-token")));
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:196]: links to be included in this test

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 4]
```diff
@@ -195,15 +218,18 @@ public void testCreateUser_withProgramTokenInUserObject() throws Exception {
...
         assertThat(apiClientUser, is(notNullValue()));
         assertThat(apiClientUser.getFirstName(), is(equalTo("test-first-name")));
         assertThat(apiClientUser.getBusinessOperatingName(), is(equalTo("test-business-operating-name")));
         assertThat(apiClientUser.getStatus(), is(nullValue()));
         assertThat(apiClientUser.getVerificationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getBusinessStakeholderVerificationStatus(), is(nullValue()));
+        assertThat(apiClientUser.getLetterOfAuthorizationStatus(), is(equalTo(LetterOfAuthorizationStatus.VERIFIED)));
+        assertThat(apiClientUser.getGovernmentIdType(), is(nullValue()));
         assertThat(apiClientUser.getCreatedOn(), is(nullValue()));
         assertThat(apiClientUser.getProgramToken(), is(equalTo("test-program-token2")));
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:234]: Pls include links in this test

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 5]
```diff
@@ -3528,6 +3497,7 @@ public void testGetPayPalAccountStatusTransition_successful() throws Exception {
         HyperwalletStatusTransition resp = client.getPayPalAccountStatusTransition("test-user-token", "test-paypal-account-token", "test-status-transition-token");
         assertThat(resp, is(equalTo(transitionResponse)));
 
+
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:3500]: Pls remove this blank line

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 6]
```diff
@@ -3740,10 +3714,12 @@ public void testCreateBankAccount_successful() throws Exception {
         assertThat(resp, is(equalTo(bankAccountResponse)));
 
         ArgumentCaptor<HyperwalletBankAccount> argument = ArgumentCaptor.forClass(HyperwalletBankAccount.class);
+
         Mockito.verify(mockApiClient)
                 .post(Mockito.eq("https://api.sandbox.hyperwallet.com/rest/v4/users/test-user-token/bank-accounts"), argument.capture(),
                         Mockito.eq(bankAccount.getClass()));
 
+
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:3722]: Blank line can be removed

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 7]
```diff
@@ -4177,9 +4167,11 @@ public void testListBankAccountStatusTransitions_noParameters() throws Exception
         HyperwalletList<HyperwalletStatusTransition> resp = client.listBankAccountStatusTransitions("test-user-token", "test-bank-account-token");
         assertThat(resp, is(equalTo(response)));
 
+
         Mockito.verify(mockApiClient).get(Mockito
                         .eq("https://api.sandbox.hyperwallet.com/rest/v4/users/test-user-token/bank-accounts/test-bank-account-token/status-transitions"),
                 Mockito.any(TypeReference.class));
+
     }
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:4176]: Extra blank lines above can be removed

**Review by arao6 — APPROVED** [review]

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java [hunk 8]
```diff
@@ -25,11 +25,20 @@
...
     public static enum Status {PRE_ACTIVATED, ACTIVATED, LOCKED, FROZEN, DE_ACTIVATED}
 
-    public static enum VerificationStatus {UNDER_REVIEW, VERIFIED, REQUIRED, NOT_REQUIRED}
+    public static enum VerificationStatus {NOT_REQUIRED, REQUIRED, FAILED, UNDER_REVIEW, VERIFIED}
+
+    public static enum BusinessStakeholderVerificationStatus {NOT_REQUIRED, REQUIRED, FAILED, UNDER_REVIEW, VERIFIED, READY_FOR_REVIEW}
+
+    public static enum LetterOfAuthorizationStatus {NOT_REQUIRED, REQUIRED, FAILED, UNDER_REVIEW, VERIFIED, READY_FOR_REVIEW}
+
+    public static enum GovernmentIdType {PASSPORT, NATIONAL_ID_CARD}
```
Comment (vkorotchenko) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java:34]: do we need to include any other id types? like drivers license etc?

**Review by vkorotchenko — APPROVED** [review]

**Pull request merged**